site/src/pages/release-notes/1.6.0.mdx

```diff
@@ -0,0 +1,315 @@
+  BodyPart part2 = BodyPart.of(ContentDisposition.of("form-data", "password"), "mypassword");
+  Multipart multipart = Multipart.of(part1, part2);
+  client.execute(multipart.toHttpRequest("/login"));
+
+  // Handle a multipart message.
+  Server.builder()
+        .service("/simple", (ctx, req) -> {
+            Multipart multipart = Multipart.from(req);
+            multipart.aggregate().thenApply(aggregated -> {
+                assert aggregated.field("username").contentUtf8().equals("Armeria");
+                assert aggregated.field("password").contentUtf8().equals("mypassword");
+                ...
+            }));
+       });
+  ```
+- Armeria now provides various useful extension and conversions for Scala. #3395
+  - See [Scala integration](/docs/advanced-scala) for the full features.
+- You can now create a <type://StreamMessage> from a `Path` or a `File`. #3344
+  ```java
+  Path path = Paths.get("...");
+  StreamMessage<HttpData> publisher = StreamMessage.of(path);
+  ```
+- You can now filter or transform values of a <type://StreamMessage> using
+  <type://StreamMessage#filter(Predicate)> or <type://StreamMessage#map(Function)>. #3351
+  ```java
+  // Filter.
+  StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+  StreamMessage<Integer> even = source.filter(x -> x % 2 == 0);
+
+  // Transform.
+  StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+  StreamMessage<Boolean> isEven = source.map(x -> x % 2 == 0);
+  ```
+- You can now send a different response depending on the exception. #3209 #3413
+  ```java
+  ExceptionHandler handler = (ctx, cause) -> {
+      if (cause instanceof RequestTimeoutException) {
+          // The request timed out!
+          return AggregatedHttpResponse.of(...);
+      }
+
+      if (cause instanceof IllegalArgumentException) {
+          return AggregatedHttpResponse.of(HttpStatus.BAD_REQUEST);
+      }
+
+      // Return null to let ExceptionHandler.ofDefault() convert the exception.
+      return null;
+  }
+
+  Server.builder().exceptionHandler(handler)...
+  ```
+- You can now convert an exception into an <type://RpcResponse> in <type://THttpService>. #3349 #3383
+  ```java
+  THttpService.builder()
+              .addService("hello", helloService)
+              .exceptionHandler((ctx, cause) -> {
+                  if (cause instanceof IllegalArgumentException) {
+                      return RpcResponse.of(CustomizedException("Bad Request!"));
+                  }
+                  ...
+              })
+  ```
+- You can now set a response timeout and attributes using <type://WebClientRequestPreparation>. #3347 #3357
+  ```java
+  WebClient client = ...;
+  client.prepare()
+        .get("/my-service")
+        .responseTimeout(Duration.ofSeconds(3))
+        .attr(USER_ID, userId)
+        .attr(USER_SECRET, secret)
+        .execute();
+  ```
+- You can now add additional error details to the response when an exception is raised in gRPC service.
```

Comment:
- add additional -> specify additional ? (not two 'add's)
- the response -> a response (or a gRPC response)
- a gRPC service